cedar-policy/src/api.rs

```diff
@@ -3071,21 +3068,10 @@ impl<S> RequestBuilder<S> {
     ///
     /// Note that you can create the `EntityUid` using `.parse()` on any
     /// string (via the `FromStr` implementation for `EntityUid`).
-    ///
-    /// Here, passing `None` for `principal` indicates that `principal` does
-    /// not contribute to authorization decisions (e.g., because it is not
-    /// used in your policies).
-    /// This is different than Unknown for partial-evaluation purposes.
     #[must_use]
-    pub fn principal(self, principal: Option<EntityUid>) -> Self {
+    pub fn principal(self, principal: EntityUid) -> Self {
         Self {
-            principal: match principal {
-                Some(p) => ast::EntityUIDEntry::concrete(p.into(), None),
-                None => ast::EntityUIDEntry::concrete(
-                    ast::EntityUID::unspecified_from_eid(ast::Eid::new("principal")),
-                    None,
-                ),
-            },
+            principal: ast::EntityUIDEntry::concrete(principal.into(), None),
```

Comment:
Sad that we're still using the "concrete" terminology. Can you change this to "known"?

Reply:
Good call